src/app/modules/submission/multirespondent-submission/__tests__/multirespondent-submission.service.spec.ts

```diff
@@ -463,9 +578,133 @@ describe('multirespondent-submission.service', () => {
       expect(sendMRFWorkflowStepEmailSpy).not.toHaveBeenCalled()
       // is rejected email and destination emails are correct
       expect(sendMrfApprovalEmailSpy.mock.calls[0][0].isRejected).toBeTrue()
-      expect(
-        sendMrfApprovalEmailSpy.mock.calls[0][0].emails,
-      ).toContainAllValues(expectedEmails)
+      expect(sendMrfApprovalEmailSpy.mock.calls[0][0].emails).toEqual(
+        expectedEmails,
+      )
+      expect(sendMrfApprovalEmailSpy.mock.calls[0][0].emails.length).toBe(
+        expectedEmails.length,
+      )
+    })
+
+    it('workflow terminates and sends not approved outcome email only to steps before and including current step when mrf is rejected for mid step of multiple step MRF', async () => {
+      // Arrange
+      const sendMrfWorkflowCompletionEmailSpy = jest.spyOn(
+        MailService,
+        'sendMrfWorkflowCompletionEmail',
+      )
+      const sendMrfApprovalEmailSpy = jest.spyOn(
+        MailService,
+        'sendMrfApprovalEmail',
+      )
+      const sendMRFWorkflowStepEmailSpy = jest.spyOn(
+        MailService,
+        'sendMRFWorkflowStepEmail',
+      )
+
+      const expectedEmails = [
+        'expected1@example.com',
+        'expected2@example.com',
+        'expected3@example.com',
+      ]
+
+      const stepOneId = new ObjectId().toHexString()
+      const stepTwoId = new ObjectId().toHexString()
+      const stepThreeId = new ObjectId().toHexString()
+      const stepFourId = new ObjectId().toHexString()
+      const stepFiveId = new ObjectId().toHexString()
+
+      const emailFieldId1 = new ObjectId().toHexString()
+      const emailFieldId2 = new ObjectId().toHexString()
+
+      const yesNoFieldId1 = new ObjectId().toHexString()
+      const yesNoFieldId2 = new ObjectId().toHexString()
+
+      const submissionResponses = {
+        [emailFieldId1]: {
+          fieldType: BasicField.Email,
+          answer: {
+            value: expectedEmails[0],
+          },
+        },
+        [emailFieldId2]: {
+          fieldType: BasicField.Email,
+          answer: {
+            value: 'not_expected_1@example.com',
+          },
+        },
+        [yesNoFieldId1]: {
+          fieldType: BasicField.YesNo,
+          answer: 'No',
+        },
+      } as FieldResponsesV3
+
+      const fourStepApprovalWorkflow: FormWorkflowStepDto[] = [
```

Comment:
Wouldn't this be regarded as 5 steps?

Reply:
yes, i added another step but forgot to update the name. Thanks for this catch!